URGENT: LoadGen API broken

https://github.com/mlcommons/inference/pull/1184 committed both to master and r2.1 has broken the LoadGen API just 10 days before to the deadline. This must now be accommodated for in submitters workflows.

Moreover, the change is incomplete: `const` must be removed from this line as well
https://github.com/mlcommons/inference/blob/r2.1/loadgen/query_sample_library.h#L37

Here are the errors I got from our [QAIC](https://github.com/krai/ck-qaic) implementation:
```
g++-9 -c    -I../ -DCK_HOST_OS_NAME2_LINUX=1 -DCK_HOST_OS_NAME_LINUX=1 -DCK_TARGET_OS_NAME2_LINUX=1 -DCK_TARGET_OS_NAME_LINUX=1 -std=c++11 -O3 -ffast-math    -Wall -Wno-sign-compare -std=c++17 -I/opt/qti-aic/dev/inc/ -I/opt/qti-aic/dev/inc/qaic-api-common -I/home/anton/CK-TOOLS/lib-qaic-api-0.3-gcc-9.4.0-master-linux-64/install/ -I/home/anton/CK-TOOLS/lib-protobuf-host-for.qaic-linux-64/install/include/ -I/home/anton/CK-TOOLS/lib-qaic-api-0.3-gcc-9.4.0-master-linux-64/include -I/home/anton/CK-TOOLS/lib-rtl-xopenme-0.3-gcc-9.4.0-linux-64/include -I/home/anton/CK-TOOLS/lib-mlperf-loadgen-static-gcc-9.4.0-compiler.python-3.8.10-linux-64/include  ../harness.cpp  -o harness.o                                                                                                                                                         In file included from ../harness.cpp:46:                                                                                                                                                                        ../harness.h:181:22: error: ‘const string& SystemUnderTestQAIC::Name() const’ marked ‘override’, but does not override                                                                                            181 |   const std::string &Name() const override { return name_; }
      |                      ^~~~
../harness.cpp: In function ‘void TestQAIC(Program*)’:
../harness.cpp:610:23: error: cannot declare variable ‘sut’ to be of abstract type ‘SystemUnderTestQAIC’
  610 |   SystemUnderTestQAIC sut(prg, ts.scenario);
      |                       ^~~
In file included from ../harness.cpp:46:
../harness.h:175:7: note:   because the following virtual functions are pure within ‘SystemUnderTestQAIC’:
  175 | class SystemUnderTestQAIC : public mlperf::SystemUnderTest {
      |       ^~~~~~~~~~~~~~~~~~~
In file included from ../harness.cpp:42:
/home/anton/CK-TOOLS/lib-mlperf-loadgen-static-gcc-9.4.0-compiler.python-3.8.10-linux-64/include/system_under_test.h:41:30: note:       ‘virtual const string& mlperf::SystemUnderTest::Name()’
   41 |   virtual const std::string& Name() = 0;
      |                              ^~~~
In file included from ../benchmark.h:62,
                 from ../harness.cpp:45:
```
We can live with this change if `system_under_test.h` is also properly updated. Otherwise, we will freeze our code at the previous commit.

--- a/loadgen/benchmark/repro.cpp
+++ b/loadgen/benchmark/repro.cpp
@@ -30,7 +30,7 @@
 class QSL : public mlperf::QuerySampleLibrary {
  public:
   ~QSL() override{};
-  const std::string& Name() const override { return mName; }
+  const std::string& Name() override { return mName; }
   size_t TotalSampleCount() override { return 1000000; }
   size_t PerformanceSampleCount() override { return TotalSampleCount(); }
   void LoadSamplesToRam(

--- a/loadgen/bindings/c_api.cc
+++ b/loadgen/bindings/c_api.cc
@@ -85,7 +85,7 @@ class QuerySampleLibraryTrampoline : public QuerySampleLibrary {
         unload_samples_from_ram_cb_(unload_samples_from_ram_cb) {}
   ~QuerySampleLibraryTrampoline() override = default;
 
-  const std::string& Name() const override { return name_; }
+  const std::string& Name() override { return name_; }
   size_t TotalSampleCount() override { return total_sample_count_; }
   size_t PerformanceSampleCount() override { return performance_sample_count_; }
 

--- a/loadgen/bindings/python_api.cc
+++ b/loadgen/bindings/python_api.cc
@@ -106,7 +106,7 @@ class QuerySampleLibraryTrampoline : public QuerySampleLibrary {
         unload_samples_from_ram_cb_(unload_samples_from_ram_cb) {}
   ~QuerySampleLibraryTrampoline() override = default;
 
-  const std::string& Name() const override { return name_; }
+  const std::string& Name() override { return name_; }
   size_t TotalSampleCount() { return total_sample_count_; }
   size_t PerformanceSampleCount() { return performance_sample_count_; }
 

--- a/loadgen/query_sample_library.h
+++ b/loadgen/query_sample_library.h
@@ -34,7 +34,7 @@ class QuerySampleLibrary {
   virtual ~QuerySampleLibrary() {}
 
   /// \brief A human readable name for the model.
-  virtual const std::string& Name() const = 0;
+  virtual const std::string& Name() = 0;
 
   /// \brief Total number of samples in library.
   virtual size_t TotalSampleCount() = 0;

